static/client/common.js

```diff
@@ -299,3 +299,53 @@ CCC.Common.escapeSpaces = function(text) {
       .replace(/  /g, '\u00A0 ').replace(/  /g, '\u00A0 ')
       .replace(/^ /gm, '\u00A0');
 };
+
+/**
+ * Detect URLs in the provided document fragment and replace with links.
+ * @param {!Element} el A DOM element to scan and modify.
+ */
+CCC.Common.autoHyperlink = function(el) {
+  var isSvg = el.namespaceURI == CCC.Common.NS;
+  var children = el.childNodes;  // This is a live NodeList.
+  for (var i = children.length - 1, child; (child = children[i]); i--) {
+    if (child.nodeType === 3) {
```

Comment:
* `Node.TEXT_NODE`.

* Why not move this test up to the top of the function, so the two loops can be separated?:

```JS
  if (el.nodeType !== Node.TEXT_NODE) {
    // split text and loop over parts.
  } else {
    for (child of …) {
      CCC.Common.autoHyperlink(child);
    }
  }
```

* Failing that, at least invert the test and use continue, to avoid unnecessary indentation:
```JS
  if (child.nodeType !== Node.TEXT_NODE) {
      CCC.Common.autoHyperlink(child);
      continue;
    }
    var text = …
```

Reply:
Changed to use Node properties.  Though this is not the JS way, as seen in every one of our dependencies.

The first option isn't a great idea since we are passing a text element, but its parent node is the one being modified.  Unexpected side effects.  I'll go with the second option.